network/network_test.go

```diff
@@ -288,42 +288,37 @@ func newFullyConnectedTestNetwork(t *testing.T, handlers []router.InboundHandler
 			}, 10*time.Second, time.Millisecond)
 		}
 
-		go func(net Network) {
-			defer wg.Done()
-
-			require.NoError(net.Dispatch())
-		}(net)
+		eg.Go(net.Dispatch)
 	}
 
 	if len(networks) > 1 {
 		<-onAllConnected
 	}
 
-	return nodeIDs, networks, &wg
+	return nodeIDs, networks, eg
 }
 
 func TestNewNetwork(t *testing.T) {
-	_, networks, wg := newFullyConnectedTestNetwork(t, []router.InboundHandler{nil, nil, nil})
+	require := require.New(t)
+	_, networks, eg := newFullyConnectedTestNetwork(t, []router.InboundHandler{nil, nil, nil})
 	for _, net := range networks {
 		net.StartClose()
 	}
-	wg.Wait()
+	require.NoError(eg.Wait())
 }
 
 func TestIngressConnCount(t *testing.T) {
 	require := require.New(t)
 
 	emptyHandler := func(context.Context, message.InboundMessage) {}
 
-	_, networks, wg := newFullyConnectedTestNetwork(
+	_, networks, _ := newFullyConnectedTestNetwork(
```

Comment:
The prior code here was definitely wrong, rather than deleting the `eg` entirely, I think we should add:
```go
for _, net := range networks {
	net.StartClose()
}
require.NoError(eg.Wait())
```
to the end of this test.